File: src/test/java/net/dontdrinkandroot/gitki/wicket/page/user/UserEditPageTest.java
package net.dontdrinkandroot.gitki.wicket.page.user;

import net.dontdrinkandroot.gitki.test.AbstractWicketTest;
import org.junit.jupiter.api.Test;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class UserEditPageTest extends AbstractWicketTest
{
    @Test
    public void testAccess()
    {
        this.assertLoginRequired(UserEditPage.class);
        this.assertPageAccessible(UserEditPage.class, this.userWatcher);
        this.assertPageAccessible(UserEditPage.class, this.userCommitter);
        this.assertPageAccessible(UserEditPage.class, this.userAdmin);
    }
}

File: src/test/kotlin/net/dontdrinkandroot/gitki/wicket/page/user/UserEditPageTest.kt
package net.dontdrinkandroot.gitki.wicket.page.user

import net.dontdrinkandroot.gitki.test.AbstractWicketTest
import org.junit.jupiter.api.Test

class UserEditPageTest : AbstractWicketTest() {

    @Test
    fun testAccess() {
        assertLoginRequired(UserEditPage::class.java)
        assertPageAccessible(UserEditPage::class.java, userWatcher)
        assertPageAccessible(UserEditPage::class.java, userCommitter)
        assertPageAccessible(UserEditPage::class.java, userAdmin)
    }
}